### PR TITLE
Added missed input nodes to Visual Shader

### PR DESCRIPTION
**Pull request by Darkinggq**
image

Unfortunately, i am not very English and i cannot help the description of the visual shader (someone will write a brief text description)
image

To make three separate description:
(Particles only)
(GLES2 only)
(GLES3 / VULKAN only)

the user reads the description easier ^_^

### CODE REVIEW

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 1]
```diff
@@ -1483,9 +1483,36 @@ VisualShaderEditor::VisualShaderEditor() {
...
+	add_options.push_back(AddOption("Custom", "Input", "Vertex", "VisualShaderNodeInput", TTR("(Particles only) 'custom' ..."), "custom", VisualShaderNode::PORT_TYPE_VECTOR));
+	add_options.push_back(AddOption("CustomAlpha", "Input", "Vertex", "VisualShaderNodeInput", TTR("(Particles only) 'custom_alpha' ..."), "custom_alpha", VisualShaderNode::PORT_TYPE_SCALAR));
+	add_options.push_back(AddOption("Delta", "Input", "Vertex", "VisualShaderNodeInput", TTR("(Particles only) 'delta' ..."), "delta", VisualShaderNode::PORT_TYPE_SCALAR));
+	add_options.push_back(AddOption("Lifetime", "Input", "Vertex", "VisualShaderNodeInput", TTR("(Particles only) 'lifetime' ..."), "lifetime", VisualShaderNode::PORT_TYPE_SCALAR));
+	add_options.push_back(AddOption("Index", "Input", "Vertex", "VisualShaderNodeInput", TTR("(Particles only) 'index' ..."), "index", VisualShaderNode::PORT_TYPE_SCALAR));
+	add_options.push_back(AddOption("EmissionTransform", "Input", "Vertex", "VisualShaderNodeInput", TTR("(Particles only) 'em_transform' ..."), "emission_transform", VisualShaderNode::PORT_TYPE_TRANSFORM));
```
Comment (Chaosus) [inline editor/plugins/visual_shader_editor_plugin.cpp]: I guess you can add the same "input parameter for x shader mode." to the descriptions - there is no complex english here.

Also dont forget to setup the shader modes - VisualShader::TYPE_VERTEX or (VisualShader::TYPE_FRAGMENT |  VisualShader::TYPE_LIGHT)

Comment (ilyafuu) [reply editor/plugins/visual_shader_editor_plugin.cpp]: okay

**Comment by Chaosus** [comment]
Dont forget to merge the commits to one as described in http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

**Comment by ilyafuu** [comment]
badly forgot one old commit impossible merge (I use gitkraken) :(

**Comment by ilyafuu** [comment]
topic can be closed no sense again to make torture.

**Pull request closed without merging**